js/src/reports/utils.test.js

```diff
@@ -0,0 +1,296 @@
+
+			expect( result ).toHaveLength( expectedIntervals.length );
+			expectedIntervals.forEach( ( item, i ) => {
+				expect( result[ i ] ).toMatchObject( item );
+			} );
+		} );
+
+		it( "should merge two given intervals' items to a union result by `interval`", () => {
+			const intervals1 = toIntervals( [ '2021-01', 12 ] );
+			const intervals2 = toIntervals( [ '2021-02', 34 ] );
+			const expectedIntervals = toIntervals(
+				[ '2021-01', 12 ],
+				[ '2021-02', 34 ]
+			);
+
+			const result = aggregateIntervals( intervals1, intervals2 );
+
+			expect( result ).toHaveLength( expectedIntervals.length );
+			expectedIntervals.forEach( ( item, i ) => {
+				expect( result[ i ] ).toMatchObject( item );
+			} );
+		} );
+
+		it( 'When any metric in `paidFields` does not exist in `subtotals`, should fill its value with 0', () => {
+			const intervals1 = toIntervals( [ '2021-01', , 1, , 3 ] );
+			const intervals2 = toIntervals( [ '2021-01', , , 2, , 4 ] );
+			const expectedIntervals = toIntervals( [
+				'2021-01',
+				0,
+				1,
+				2,
+				3,
+				4,
+			] );
+
+			const result = aggregateIntervals( intervals1, intervals2 );
+
+			expectedIntervals.forEach( ( item, i ) => {
+				expect( result[ i ] ).toMatchObject( item );
+			} );
+		} );
+	} );
+} );
+
+describe( 'sumToPerformance', () => {
+	function toTotals( fields, ...values ) {
+		return values.reduce(
+			( acc, value, i ) => ( {
+				...acc,
+				[ fields[ i ] ]: value,
+			} ),
+			{}
+		);
+	}
+
+	const toFreeTotals = toTotals.bind( null, freeFields );
+	const toPaidTotals = toTotals.bind( null, paidFields );
+
+	it( 'should be able to sum paid `totals` to performance only', () => {
+		const paid = toPaidTotals( 123 );
+		const free = undefined;
+		const performance = sumToPerformance( paid, free );
+
+		expect( performance ).toMatchObject( {
+			[ paidFields[ 0 ] ]: {
+				value: 123,
+				delta: null,
+				missingFreeListingsData: MISSING_FREE_LISTINGS_DATA.NONE,
+			},
+		} );
+	} );
+
+	it( 'should be able to sum free `totals` to performance only', () => {
+		const paid = undefined;
+		const free = toFreeTotals( 456 );
+		const performance = sumToPerformance( paid, free, freeFields );
+
+		expect( performance ).toMatchObject( {
+			[ freeFields[ 0 ] ]: {
+				value: 456,
+				delta: null,
+				missingFreeListingsData: MISSING_FREE_LISTINGS_DATA.NONE,
+			},
+		} );
+	} );
+
+	it( 'should sum the same field from paid `totals` and free `totals` to performance', () => {
+		const sameFields = [ 'ranks' ];
+		const paid = toTotals( sameFields, 123 );
+		const free = toTotals( sameFields, 456 );
+		const performance = sumToPerformance( paid, free, sameFields );
+
+		expect( performance ).toMatchObject( {
+			[ sameFields[ 0 ] ]: {
+				value: 579,
+				delta: null,
+				missingFreeListingsData: MISSING_FREE_LISTINGS_DATA.NONE,
+			},
+		} );
+	} );
+
+	it( 'When a paid field is not (yet) available in API, should flag the data is not available', () => {
```

Comment:
:nail_care: This test and the one bellow, seems to be a special case of the one above? WDYT of grouping them in a `describe` so it could be easier to collapse it, and browse through the list of all specs.

For example, to something like:
```js
it( 'When paid and free `totals` are given', () => {
	it( 'should sum the same fields from paid and free `totals` to performance' );
	it( 'When a paid field is not (yet) available in API, should flag the data is not available' );
	it( `When an expected free field doesn't exist, should flag anticipated data is not returned from API` );
} );
```

Reply:
Added a group by https://github.com/woocommerce/google-listings-and-ads/pull/866/commits/09ac902c6f757a67ff1d266588d6ba9387c6f8a2